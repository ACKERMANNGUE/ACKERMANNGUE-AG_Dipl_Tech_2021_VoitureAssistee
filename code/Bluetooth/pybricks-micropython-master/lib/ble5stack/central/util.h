#ifndef UTIL_H
#define UTIL_H

#include "hci_tl.h"

#define HCI_UTIL_UNKNOWN1               0xFE87
#define HCI_UTIL_GET_LEGO_FW_VERSION    0xFE88

HCI_StatusCodes_t Util_readLegoHwVersion(void);
HCI_StatusCodes_t Util_readLegoFwVersion(void);

#endif // UTIL_H
