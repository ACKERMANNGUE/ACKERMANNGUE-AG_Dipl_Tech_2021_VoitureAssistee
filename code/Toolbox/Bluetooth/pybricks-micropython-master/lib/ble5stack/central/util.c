
#include "hci_tl.h"
#include "util.h"


HCI_StatusCodes_t Util_readLegoHwVersion(void)
{
    return HCI_sendHCICommand(HCI_UTIL_UNKNOWN1, NULL, 0);
}

HCI_StatusCodes_t Util_readLegoFwVersion(void)
{
    return HCI_sendHCICommand(HCI_UTIL_GET_LEGO_FW_VERSION, NULL, 0);
}

