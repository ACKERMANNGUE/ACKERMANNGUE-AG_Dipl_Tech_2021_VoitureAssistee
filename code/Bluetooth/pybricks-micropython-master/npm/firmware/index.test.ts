// SPDX-License-Identifier: MIT
// Copyright (c) 2020 The Pybricks Authors

import * as fs from 'fs';
import * as path from 'path';
import * as util from 'util';
import { FirmwareReader, FirmwareReaderErrorCode } from '.';

const readFile = util.promisify(fs.readFile);

test('bad zip data', async () => {
    try {
        await FirmwareReader.load(new Uint8Array(100));
        fail('FirmwareReader.load() should have failed');
    } catch (err) {
        expect(err.name).toMatch('FirmwareReaderError');
        expect(err.code).toBe(FirmwareReaderErrorCode.ZipError);
    }
});

test('missing firmware-base.bin', async () => {
    var file = await readFile(
        path.resolve(__dirname, '__tests__', 'movehub-no-firmware-base.zip')
    );
    try {
        await FirmwareReader.load(file);
        fail('FirmwareReader.load() should have failed');
    } catch (err) {
        expect(err.name).toMatch('FirmwareReaderError');
        expect(err.code).toBe(FirmwareReaderErrorCode.MissingFirmwareBaseBin);
    }
});

test('missing firmware.metadata.json', async () => {
    var file = await readFile(
        path.resolve(__dirname, '__tests__', 'movehub-no-metadata.zip')
    );
    try {
        await FirmwareReader.load(file);
        fail('FirmwareReader.load() should have failed');
    } catch (err) {
        expect(err.name).toMatch('FirmwareReaderError');
        expect(err.code).toBe(FirmwareReaderErrorCode.MissingMetadataJson);
    }
});

test('missing main.py', async () => {
    var file = await readFile(
        path.resolve(__dirname, '__tests__', 'movehub-no-main-py.zip')
    );
    try {
        await FirmwareReader.load(file);
        fail('FirmwareReader.load() should have failed');
    } catch (err) {
        expect(err.name).toMatch('FirmwareReaderError');
        expect(err.code).toBe(FirmwareReaderErrorCode.MissingMainPy);
    }
});

test('missing ReadMe_OSS.txt', async () => {
    var file = await readFile(
        path.resolve(__dirname, '__tests__', 'movehub-no-readme-oss.zip')
    );
    try {
        await FirmwareReader.load(file);
        fail('FirmwareReader.load() should have failed');
    } catch (err) {
        expect(err.name).toMatch('FirmwareReaderError');
        expect(err.code).toBe(FirmwareReaderErrorCode.MissingReadmeOssTxt);
    }
});

test('reading data works', async () => {
    var file = await readFile(
        path.resolve(__dirname, '__tests__', 'movehub.zip')
    );
    var reader = await FirmwareReader.load(file);
    expect(await reader.readFirmwareBase()).toMatchSnapshot();
    expect(await reader.readMetadata()).toMatchSnapshot();
    expect(await reader.readMainPy()).toMatchSnapshot();
    expect(await reader.readReadMeOss()).toMatchSnapshot();
});
