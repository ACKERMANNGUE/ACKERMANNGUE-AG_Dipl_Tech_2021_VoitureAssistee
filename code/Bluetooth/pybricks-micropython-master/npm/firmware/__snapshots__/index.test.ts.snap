// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reading data works 1`] = `
Uint8Array [
  0,
  64,
  0,
  32,
  197,
  240,
  0,
  8,
  17,
  241,
  0,
  8,
  17,
  241,
  0,
  8,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  17,
  241,
  0,
  8,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  17,
  241,
  0,
  8,
  137,
  50,
  1,
  8,
  17,
  241,
  0,
  8,
  0,
  0,
  0,
  0,
  17,
  241,
  0,
  8,
  17,
  241,
  0,
  8,
  17,
  241,
  0,
  8,
  229,
  51,
  1,
  8,
  237,
  43,
  1,
  8,
  17,
  241,
  0,
  8,
  0,
  0,
  0,
  0,
]
`;

exports[`reading data works 2`] = `
Object {
  "checksum-type": "sum",
  "device-id": 64,
  "firmware-version": "v3.0.0a10",
  "max-firmware-size": 108544,
  "metadata-version": "1.0.0",
  "mpy-abi-version": 5,
  "mpy-cross-options": Array [
    "-mno-unicode",
  ],
  "user-mpy-offset": 100008,
}
`;

exports[`reading data works 3`] = `
"from pybricks import version
from pybricks.tools import wait

print(version)
wait(5000)
print(\\"Hello, World!\\")
"
`;

exports[`reading data works 4`] = `
"========================================================================================================================

pybricks-micropython-movehub-firmware

------------------------------------------------------------------------------------------------------------------------

========================================================================================================================

 MAIN LICENSES

------------------------------------------------------------------------------------------------------------------------

MIT

MIT License Copyright (c) <year> <copyright holders>

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the \\"Software\\"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice (including the next paragraph) shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED \\"AS IS\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

------------------------------------------------------------------------------------------------------------------------
"
`;
